project/frontend/src/App.js

```diff
@@ -36,57 +41,57 @@ function App() {
     setCompId(parsedCompId);
   }, []);
 
-  if (user.signedIn && !user.id) {
-    //TODO: fetch for id... with useremail param... once backend is ready
-    //for now id hardcoded to 123
-    setUser((prevState) => {
+  const fetchCompetitionInfo = async (fetchCall, stateKey) => {
+    try {
+      const data = await fetch(fetchCall).then((response) => response.json());
+
       return {
-        ...prevState,
-        id: 1,
+        [stateKey]: data,
       };
-    });
-  }
-
-  const fetchAndUpdateCompetitionInfo = async (fetchCall, stateKey) => {
-    return fetch(fetchCall)
-      .then((response) => response.json())
-      .then((data) => {
-        setCompetitionInfo((prevState) => {
-          return {
-            ...prevState,
-            [stateKey]: data,
-          };
-        });
-      });
+    } catch (e) {
+      return {
+        [stateKey]: undefined,
+      };
+    }
   };
 
   React.useEffect(() => {
+    if (!authStateReceived) {
+      return;
+    }
+
     if (user.signedIn && user.id && compId) {
-      Promise.all([
-        setLoading(true),
-        fetchAndUpdateCompetitionInfo(
-          "./competitionInfo?user=" + user.id + "&competition=" + compId,
-          "generalInfo"
-        ),
-        fetchAndUpdateCompetitionInfo(
-          "./recentBuys?competition=" + compId,
-          "recentBuys"
-        ),
-        fetchAndUpdateCompetitionInfo(
-          "./investments?user=" + user.id + "&competition=" + compId,
-          "investments"
-        ),
-        fetchAndUpdateCompetitionInfo("./trending", "trending"),
-      ])
-        .then(() => {
+      setLoading(true),
+        Promise.all([
+          fetchCompetitionInfo(
+            "./competitionInfo?user=" + user.id + "&competition=" + compId,
```

Comment:
Prefer to use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) for these

Reply:
Done.